sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCasts.scala

```diff
@@ -83,10 +83,15 @@ object CollationTypeCasts extends TypeCoercionRule {
       val Seq(newInput, newDefault) = collateToSingleType(Seq(input, default))
       framelessOffsetWindow.withNewChildren(Seq(newInput, offset, newDefault))
 
+    case mapCreate : CreateMap if mapCreate.children.size % 2 == 0 =>
```

Comment:
and what do we do if `mapCreate.children.size` is odd? should we add a test to reflect this behaviour, and possible a comment here explaining what happens